src/dmd/root/array.d

```diff
@@ -137,12 +143,30 @@ public:
                 if (nentries > increment)       // if 1.5 is not enough
                     increment = nentries;
                 const allocdim = length + increment;
-                auto p = cast(T*)mem.xrealloc(data.ptr, allocdim * T.sizeof);
+                debug (stomp)
+                {
+                    // always move using allocate-copy-stomp-free
+                    auto p = cast(T*)mem.xmalloc(allocdim * T.sizeof);
+                    memcpy(p, data.ptr, length * T.sizeof);
+                    memset(data.ptr, 0xFF, data.length * T.sizeof);
+                    mem.xfree(data.ptr);
+                }
+                else
+                    auto p = cast(T*)mem.xrealloc(data.ptr, allocdim * T.sizeof);
                 data = p[0 .. allocdim];
             }
-            if (mem.isGCEnabled)
-                if (length + nentries < data.length)
-                    memset(data.ptr + length + nentries, 0, (data.length - length - nentries) * T.sizeof);
+
+            debug (stomp)
+            {
+                if (length < data.length)
+                    memset(data.ptr + length, 0xFF, (data.length - length) * T.sizeof);
+            }
+            else
+            {
+                if (mem.isGCEnabled)
+                    if (length < data.length)
+                        memset(data.ptr + length, 0xFF, (data.length - length) * T.sizeof);
```

Comment:
There is a small change for the non-stomping case here: the existing implementation assumes that the reserved space will be overwritten by the caller anyway, so it would not have to be reset here. It usually won't make a difference but for inserting large arrays, but is this change deliberate?

Reply:
ping @WalterBright

Reply:
Yes. What if the caller doesn't overwrite it? The stomping code doesn't need to be efficient, as it is only enabled for debugging porpoises.

Reply:
> The stomping code doesn't need to be efficient, as it is only enabled for debugging porpoises.

This is not debug code, but code executed if the GC is enabled.